supervisor/core/supervisor_test.go

```diff
@@ -0,0 +1,74 @@
+package core
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Test the basic Supervisor functionality.
+func TestSupervisorBase(t *testing.T) {
+	assert := assert.New(t)
+	// Create config for Supervisor.
+	cfg := new(Cfg)
+	cfg.InstancesDir = "../../test_instances"
+	cfg.TermTimeout = 100 * time.Millisecond
+
+	// Create Supervisor.
+	sv := NewSupervisor(cfg)
+
+	// Set a cleanup callback.
+	t.Cleanup(func() { sv.StopAllInstance() })
+
+	// Run several instances (some with additional env and some without).
+	instName := "test_instance"
+
+	id1, err := sv.StartInstance(instName, nil, false)
+	assert.Nilf(err, `Can't start the Instance.Error: "%v"`, err)
+
+	id2, err := sv.StartInstance(instName, []string{"INSTSIGIGNORE=true"}, false)
+	assert.Nilf(err, `Can't start the Instance.Error: "%v"`, err)
+
+	_, err = sv.StartInstance(instName, nil, false)
+	assert.Nilf(err, `Can't start the Instance. Error: "%v"`, err)
+
+	_, err = sv.StartInstance(instName, []string{"INSTSIGIGNORE=true"}, false)
+	assert.Nilf(err, `Can't start the Instance. Error: "%v"`, err)
+
+	// We need to wait for the new process to set handlers.
+	// It is necessary to update for more correct synchronization.
+	time.Sleep(100 * time.Millisecond)
+
+	// Check  all instance are running.
+	assert.Equal(len(sv.ListInstances()), 4,
+		"Not all instances have been started.")
+
+	// Check the functionality of "StatusInstance".
+	status, err := sv.StatusInstance(id2)
+	assert.Nilf(err, `Can't get Instance status. Error: "%v"`, err)
+
+	assert.True(
+		status.State == "running" && status.Env[0] == "INSTSIGIGNORE=true",
+		"Status of Instance is not correct.")
+
+	// Check the functionality of "StopInstance".
+	assert.Nilf(sv.StopInstance(id1, false),
+		`Can't stop the Instance. Error: "%v"`, err)
+	assert.Equal(len(sv.ListInstances()), 3, "Can't stop the Instance.")
+
+	// Check "StopInstance" on the Instance with "INSTSIGIGNORE=true".
+	assert.NotNil(sv.StopInstance(id2, false),
```

Comment:
I like this approach more:

```go
err = sv.StopInstance(id2, false)
assert.NotNil(err, ...)
```

Reply:
I don't see a reason to introduce an additional variable just to check the result of the method.